src/components/layerpublisher/LayerPublisher.js

```diff
@@ -24,43 +43,56 @@ function info(fileName) {
   PubSub.publish(STATUS_INFO, `Loaded ${fileName}.`);
 }
 
+function fetchDataFromDZI(layerType, data) {
+  return fetch(data[layerType].tileSource).then(response => response.text())
+    .then(str => (new window.DOMParser()).parseFromString(str, 'text/xml'))
+    .then((layer) => {
+      const { tileSource } = data[layerType];
+      data[layerType].tileSource = tileSource.substring(0, tileSource.lastIndexOf('/')); // eslint-disable-line no-param-reassign
+      // eslint-disable-next-line no-param-reassign
+      data[layerType].height = layer.getElementsByTagName('Size')[0].attributes.Height.value;
+      // eslint-disable-next-line no-param-reassign
+      data[layerType].width = layer.getElementsByTagName('Size')[0].attributes.Width.value;
+      // eslint-disable-next-line no-param-reassign
+      data[layerType].tileSize = layer.getElementsByTagName('Image')[0].attributes.TileSize.value;
+      return data;
+    });
+}
+
+function fetchImageMetadata(data) {
+  return fetchDataFromDZI('nuclei', data).then(res => fetchDataFromDZI('polyT', res));
```

Comment:
We really shouldn't hardcode these... could a list be included in `data`? Or failing that, maybe we fetch a file with a list first? Also, unless there's a reason they need to be sequential, look at [`all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all).

Reply:
Sounds good